### DISCUSSION TITLE
MLFlow logger step vs epoch

### DISCUSSION THREAD

**Discussion by JackCaster** [Lightning-AI/pytorch-lightning#7463] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I am using [MLFlowLogger](https://pytorch-lightning.readthedocs.io/en/stable/extensions/generated/pytorch_lightning.loggers.MLFlowLogger.html#pytorch_lightning.loggers.MLFlowLogger) to log my experiment into Azure ML. Everything works fine but I noticed that when I ask the logger to store a metric every step (instead of every epoch), the logger does not increase the step number but instead keeps overwriting the current step in the batch. That is, if I have 5 minibatches for each epoch, and I have 10 epochs, the logger will overwrite step 1-5 continuously, instead of logging step 1-50. Something I noticed is that MLFlow/AzureML log metrics according to steps, regardless of whether it is a epoch or not, perhaps this is causing some issues.

Would you be able to help?

An example here (7 steps only for `train_loss` but I ran more than 70 epochs)
![image](https://user-images.githubusercontent.com/18400141/117634100-e6bbe000-b17e-11eb-991f-5985e58278b7.png)
![image](https://user-images.githubusercontent.com/18400141/117635314-fee02f00-b17f-11eb-9ce5-703b36debeb0.png)

**Comment by JackCaster** [comment, marked as answer]
I got it. By default the logger logs every 50 steps (https://pytorch-lightning.readthedocs.io/en/1.3.0/extensions/logging.html#control-logging-frequency). It so happened that my dataset was very small so there were way more epochs than training steps. If I increase the frequency `trainer = Trainer(log_every_n_steps=1)` then I do get a log for every steps in the training process. 
![image](https://user-images.githubusercontent.com/18400141/117647064-cbf06800-b18c-11eb-9ac8-354eb5d285db.png)

**Reply by Kalyankr** [reply]
Hi @JackCaster , I am trying to log metric to AzureML by running as an experiment and i have problems in using the MLflowLogger, is there any sample code or notebook that can help me understand how i can use the MLFlowLogger to log in AzureML

You help is much appreciated, thank you
